assign_rights/assemble.py

```diff
@@ -21,3 +25,12 @@ def create_json(self):
     def return_rights(self):
         """docstring for return_rights"""
     pass
+
+    def run(self, rights_ids, end_date):
+        shells = []
+        for identifier in rights_ids:
+            try:
+                shells.append(self.retrieve_rights(identifier))
```

Comment:
Based on my comment above, if you opt to go that route, this function would then need to be amended as well:
```suggestion

    def run(self, rights_ids, end_date):
            try:
                rights_shells = self.retrieve_rights(rights_ids))
                for shell in rights_shells:
                    ...
```